src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -126,58 +126,107 @@ inline data_types element_type_to_data_type(ov::element::Type t) {
     }
 }
 
+constexpr size_t SHAPE_RANK_MAX = 9;
+
 /// @brief Represents data padding information.
 struct padding {
     /// @brief Filling value for padding area.
     float filling_value() const { return _filling_value; }
 
     /// @brief Gets lower padding sizes. For spatials, it means size of left (X) and top (Y) padding.
     /// @return Tensor with padding for top/left/lower bounds of data.
-    tensor lower_size() const { return _lower_size; }
+    std::vector<tensor::value_type> lower_size(const size_t max_size = 0) const {  // TODO: get shape size
+        std::vector<tensor::value_type> sizes(max_size == 0 ? sizeof(_lower_size) / sizeof(_lower_size[0]) : max_size, 0);
+        for (size_t i = 0; i < sizes.size(); ++i)
+            sizes[i] = _lower_size[i];
+        return sizes;
+    }
 
     /// @brief Gets upper padding sizes. For spatials, it means size of right (X) and bottom (Y) padding.
     /// @return Tensor with padding for bottom/right/upper bounds of data.
-    tensor upper_size() const { return _upper_size; }
+    std::vector<tensor::value_type> upper_size(const size_t max_size = 0) const {
+        std::vector<tensor::value_type> sizes(max_size == 0 ? sizeof(_upper_size) / sizeof(_upper_size[0]) : max_size, 0);
+        for (size_t i = 0; i < sizes.size(); ++i)
+            sizes[i] = _upper_size[i];
+        return sizes;
+    }
 
-    void set_dynamic_pad(const tensor& dynamic_pad_dims) {
-        _dynamic_pad_dims = dynamic_pad_dims;
+    void set_dynamic_pad_dims(const std::vector<tensor::value_type>& other) {
+        OPENVINO_ASSERT(other.size() <= SHAPE_RANK_MAX, "vector size exceeds maximum rank!");
+        const auto cnt = std::min(other.size(), SHAPE_RANK_MAX);       
+        std::copy_n(std::begin(other), cnt, _dynamic_pad_dims);
+        std::fill_n(_dynamic_pad_dims + cnt, SHAPE_RANK_MAX - cnt, 0);
     }
 
-    tensor get_dynamic_pad_dims() const {
-        return _dynamic_pad_dims;
+    std::vector<tensor::value_type> get_dynamic_pad_dims(const size_t max_size = 0) const {
+        std::vector<tensor::value_type> sizes(max_size == 0 ? sizeof(_dynamic_pad_dims) / sizeof(_dynamic_pad_dims[0]) : max_size, 0);
+        for (size_t i = 0; i < sizes.size(); ++i)
+            sizes[i] = _dynamic_pad_dims[i];
+        return sizes;
     }
+
     /// @brief
     /// @param lower_sizes Top-left padding sizes. See @ref tensor::tensor(const std::vector<value_type>&, value_type) for details.
     /// @param upper_sizes Bottom-right padding sizes. See @ref tensor::tensor(const std::vector<value_type>&, value_type) for details.
     /// @param filling_value Filling value for padding area.
     padding(const std::vector<tensor::value_type>& lower_sizes,
```

Comment:
lower_sizes and upper_sizes, as well as dynamic_pad_dims are vectors in logical format order. This is semantically different from tensor. This discrepancy leads to bugs and need fixes where padding object is initiated. 
Do you agree with this semantics change? @vladimir-paramuzov

Reply:
Yes, we need to align order of paddings with order of dimensions to avoid all these meaningless shape/pad order transformations all around